src/Compilers/Core/Portable/CodeGen/PrivateImplementationDetails.cs

```diff
@@ -146,8 +163,23 @@ private Cci.ITypeReference GetStorageStruct(uint size)
             return new ExplicitSizeStruct(size, this, _systemValueType);
         }
 
+        internal Cci.IFieldReference GetMVID(Cci.ITypeReference mvidType)
+        {
+            if (_mvidField == null)
+            {
+                Interlocked.CompareExchange(ref _mvidField, new MvidField("MVID", this, mvidType), null);
+            }
+
+            return _mvidField;
+        }
 
-        // Add a new synthesized method indexed by it's name if the method isn't already present.
+        internal Cci.IMethodDefinition StaticConstructor
+        {
+            get { return _staticConstructor; }
+            set { Interlocked.CompareExchange(ref _staticConstructor, value, null); }
```

Comment:
No, it's supposed to have no effect except the first time. The call is protected by a null check, so a second call will be uncommon, but it can legitimately happen.